### PR TITLE
Update the TopologyManager interfaces

### PR DESCRIPTION
**Pull request by klueska**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test

/kind feature

> /kind flake

**What this PR does / why we need it**:
This PR updates the core `TopologyManager` interfaces. These updates are based on discussions had about the preferred semantics of the `TopologyManager` and will be reflected in changes to PR https://github.com/kubernetes/kubernetes/pull/73580 that adds the actual `TopologyManager` implementation.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/cm/topologymanager/topology_manager.go [hunk 1]
```diff
@@ -39,15 +39,18 @@ type Manager interface {
...
 }
 
 //Store interface is to allow Hint Providers to retrieve pod affinity
 type Store interface {
 	GetAffinity(podUID string, containerName string) TopologyHint
 }
 
-// TopologyHint is a struct containing Socket Mask for a Pod
+// TopologyHint is a struct containing a SocketMask for resources allocated to
+// a container and a bool indicating whether this SocketMask encodes a
```
Comment (ConnorDoyle) [inline pkg/kubelet/cm/topologymanager/topology_manager.go]: Could we move detailed comments on field semantics to directly precede the field declaration?

e.g.

```go
// Preferred is set when...
Preferred bool
```

**Review by ConnorDoyle — CHANGES_REQUESTED** [review]

**Comment by derekwaynecarr** [comment]
This is a nice clean-up.

/approve
/lgtm

**Pull request merged**